backend/rates/history.go

```diff
@@ -169,33 +169,31 @@ func (updater *RateUpdater) backfillHistory(ctx context.Context, coin, fiat stri
 
 		end := updater.HistoryEarliestTimestamp(coin, fiat)
 		var start time.Time
-		if end.IsZero() {
-			// First time; don't have historical data yet.
+		switch {
+		// First time; don't have historical data yet.
+		case end.IsZero():
 			end = time.Now()
-			// 90 days is the max interval which CoinGecko responds to with hourly timeseries.
-			// Make sure we get hourly rates by requesting 90 days minus 1 hour, just in case.
 			start = end.Add(-90*24*time.Hour + time.Hour)
-		} else {
-			// End at the day earlier than the last known timestamp.
-			// We want daily rates for timestamps past the first 90 days: use annual intervals.
+		// 90 days is the max interval which CoinGecko responds to with hourly timeseries.
+		// Make sure we get hourly rates by requesting 90 days minus 1 hour, just in case.
+		case time.Since(end) < 90*24*time.Hour:
```

Comment:
i don't understand this condition. why is it relevant how much time is between `end` and now? we are backfilling the history, not forward filling it.

Reply:
We want hourly rates for the past 90 days, i.e. between now and (now - 90days). Suppose `end` happened to be somewhere between `now` and `90days ago`. Without this condition, it would go to the default below which requests a much larger range for daily rates but we want hourly for (now-end).

In other words:

```
(years-ago) --- daily rates --- (now-90days) --- hourly rates --- (now)
```

if `end` happens to be between `(now-90days)` and `now`, we still want the hourly rates, specifically for the range of (`end` - `now-90days`).

Reply:
I see, but why is `start = end.Add(-time.Since(end))`? If `end` is e.g. `now - 1month`, then start would be `now - 2months`? Shouldn't start be `now - 3 months` or possibly `end - 3 months`?

Reply:
oh, you're totally right and I'm a goofy :)
done.